cmake/environmentchecks.cmake

```diff
@@ -143,7 +143,7 @@ if(CLANG_FORMAT_EXE AND GIT_EXE)
   execute_process(COMMAND ${CLANG_FORMAT_EXE} -version
                   OUTPUT_VARIABLE CLANG_FORMAT_VERSION
                   ERROR_QUIET OUTPUT_STRIP_TRAILING_WHITESPACE)
-  string(REGEX REPLACE "clang-format version ([.0-9]+).*" "\\1" CLANG_FORMAT_VERSION "${CLANG_FORMAT_VERSION}")
+  string(REGEX REPLACE "^.*clang-format version ([.0-9]+).*" "\\1" CLANG_FORMAT_VERSION "${CLANG_FORMAT_VERSION}")
```

Comment:
for completeness, shouldn't the regex be `^.*clang-format version ([.0-9]+).*$` in case there is some installation that adds text to the end of the version string?

Reply:
All you added is `$` to mark the end of the string, but this is not needed as such since there is already `.*`. This being said, I put `^`, so we probably ought to be consistent, hence commit 7ff197ea.